src/main/scala/centaur/api/CromwellClient.scala

```diff
@@ -33,6 +35,19 @@ object CromwellClient {
   final implicit val materializer: ActorMaterializer = ActorMaterializer(ActorMaterializerSettings(system))
   final val apiPath = "/api/workflows/v1"
 
+  def decodeResponse(response: HttpResponse): HttpResponse = {
+    val decoder = response.encoding match {
+      case HttpEncodings.gzip ⇒
+        Gzip
+      case HttpEncodings.deflate ⇒
+        Deflate
```

Comment:
The "should we compress and if yes how" logic is actually already built-in HTTP as the requests should contain an "Accept-Encoding" header to tell the server which type of encoding it's ready to handle for the response.

Spray handles this logic itself with the compressResponse method http://spray.io/documentation/1.2.4/spray-routing/encoding-directives/compressResponse/

It happens to be that akka http doesn't do automatic decoding of requests: http://doc.akka.io/docs/akka-http/10.0.4/scala/http/common/de-coding.html
So we have to do it manually in centaur.

About should we compress all endpoints, it seemed to me that metadata was the only one that would really benefit from it and that it would be more of an overhead for others, but if we want to be consistent we could always encode them all yes.